### DISCUSSION TITLE
Multiple Module Messages

### DISCUSSION THREAD

**Discussion by SSzretter** [oqtane/oqtane.framework#4700] [General]
Is it possible to have multiple module messages?
I have found that if I call: AddModuleMessage("1", MessageType.Info);
and then another point in the code call:  AddModuleMessage("2", MessageType.Info);
It only displays "2".

I know I can accomplish this using an 3rd party component such as syncfusion or similar, just wondering if it is possible stock or this might be a good feature request...

**Comment by sbwalker** [comment]
@SSzretter the challenge is that Blazor is a stateful framework (ie. the page is not refreshed on every navigation). The current behavior is that AddModuleMessage() always initializes the message value when it is called. This behavior means that a component does not need to explicitly call ClearModuleMessage() to remove the old state. We could not modify the existing method as we would not want to introduce a breaking behavioral change. So this would require a new method ie. AppendModuleMessage() and the developer would be responsible for explicitly calling ClearModuleMessage() in their code to reset the state.
